### PR TITLE
Log avoid leaking full path

### PR DESCRIPTION
**Pull request by clementperon**
at the moment the Logger use __FILE__ that would get the complete path of the file.

```
$> strings libpcapplusplus.a | grep "\.cpp"
/tmp/PcapPlusPlus-23.09/Pcap++/src/PcapFileDevice.cpp
/tmp/PcapPlusPlus-23.09/Pcap++/src/PcapDevice.cpp
/tmp/PcapPlusPlus-23.09/Pcap++/src/PcapFilter.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/IcmpLayer.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/IPReassembly.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/IPv4Layer.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/Layer.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/Packet.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/PacketUtils.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/RawPacket.cpp
/tmp/PcapPlusPlus-23.09/Packet++/src/Sll2Layer.cpp
```

Instead use __FILE_NAME__ when available: GCC > 12 or Clang

I have also move this out of the namespace as it's a global macro and refactor introduce a PCPP_LOG(level, message) macro.

### CODE REVIEW

**Review by egecetin — APPROVED** [review]

File: Common++/header/Logger.h [hunk 1]
```diff
@@ -11,6 +11,33 @@
...
+		{ \
+			PCPP_LOG(pcpp::Logger::Debug, message); \
+		} \
+	} while(0)
+
+#define PCPP_LOG_ERROR(message) do \
+	{ \
+		PCPP_LOG(pcpp::Logger::Error, message); \
+	} while (0)
+
```
Comment (seladb) [inline Common++/header/Logger.h:40]: Why did you move these macros to the top of the file?

Comment (clementperon) [reply Common++/header/Logger.h:40]: I think it was not intuitive to put them in the namespace where they don't believe.

Comment (seladb) [reply Common++/header/Logger.h:40]: ok 👍

**Review by seladb — APPROVED** [review]
